test/support/utils.ex

```diff
@@ -4,7 +4,37 @@ defmodule NextLS.Support.Utils do
   import ExUnit.Callbacks
   import GenLSP.Test
 
-  def mix_exs do
+  def write_project(proj_name, ctx \\ %{}) do
+    tmp_dir = Map.fetch!(ctx, :tmp_dir) 
+    proj_dir = Path.join(tmp_dir, proj_name)
+    File.mkdir_p!(Path.join(proj_dir, "lib"))
+    File.write!(Path.join(proj_dir, "mix.exs"), mix_exs(ctx))
+
+    new_ctx = %{cwd: tmp_dir, proj_dir: proj_dir}
+
+    if Map.get(ctx, :with_dependency) do
+      ext = Path.join(proj_dir, "lib/ext.ex")
+
+      File.write!(ext, """
+      defmodule Ext do
+        def run() do
+          Jason.encode!(%{})
+        end
+      end
+      """)
+     
+     {msg, exit_code} = System.cmd("mix", ["deps.get"], cd: proj_dir, into: "")
```

Comment:
this can probably be in it's own function of `mix_deps_get` and then can be called by the tests as `setup :mix_deps_get`

Reply:
including the assert below this line